docs/design/AgentRun-CLI-Design.md

```diff
@@ -0,0 +1,658 @@
+
+This CLI will serve as a foundational tool enabling:
+- Rapid prototyping and local iteration of agents
+- Seamless transitions from development to staging and production environments powered by AgentCube
+- Native integration with Kubernetes clusters and cloud services for scalable deployment
+- Version control and reproducibility of agent configurations and runtime environments
+
+By abstracting away operational overhead and providing a consistent interface, the CLI lowers the barrier to entry for agent development on AgentCube. It empowers developers to focus on innovation rather than infrastructure, fosters best practices, and accelerates collaboration across the open-source community.
+## Use Case 1: Kickstart Agent Development and Publish to AgentCube
+A developer wants to create a new agent from scratch and publish it to AgentCube for public access or team collaboration. The CLI supports agents built with any framework, offering a standardized workflow for packaging, building, and publishing. 
+
+After completing development, the CLI enables the following steps:
+1. `agentrun pack -f ./` Packages the agent source code and runtime metadata into a structured workspace directory, preparing it for image creation
+2. `agentrun build -f ./` Builds a container image from the workspace, compatible with AgentCube’s Kubernetes-based runtime environment
+3. `agentrun publish -f /` Publishes the built agent image to AgentCube, making it available for invocation, sharing, and collaboration
+
+Use Case 2: Check Published Agent Status
+After publishing an agent to AgentCube, a developer may want to verify that the agent is fully registered and ready for use. The CLI provides a simple status check command:
+
+```agentrun status -f ./```
+
+This command queries AgentCube for the current state of the agent associated with the workspace. It returns key information such as the agent ID, endpoint URL, latest version, and log reference. This helps ensure the agent is correctly deployed and ready for invocation.
+
+## Use Case 3: Invoke published Agent
+After publishing an agent to AgentCube, a developer may want to invoke it for testing purposes or integrate it into other system components. The CLI provides a simple and consistent interface to trigger agent execution using the local workspace directory and a structured payload:
+
+```agentrun invoke -f ./ -payload {"prompt": "what is the weather today in Shanghai?"}```
+
+# Scope
+In Scope:
+* support agent packaging and workspace generation
+* support agent image build for Kubernetes runtime
+* support agent publishing to AgentCube
+* support agent invocation via CLI
+* support Python SDK for CI/CD integration
+# Function Detail
+### 1. Pack, Build and Publish 
+**Step 0:** The developer creates the agent application using any preferred framework and defines the required runtime metadata. The agent must expose an HTTP interface to support standardized invocation. The metadata should include:
```

Comment:
> The agent must expose an HTTP interface to support standardized invocation.

This is not what we discussed today. We are talking about standard mcp a2a protocol